sql/core/src/main/scala/org/apache/spark/sql/DataFrameNaFunctions.scala

```diff
@@ -395,9 +395,9 @@ final class DataFrameNaFunctions private[sql](df: DataFrame) {
 
   private def fillMap(values: Seq[(String, Any)]): DataFrame = {
     // Error handling
-    values.foreach { case (colName, replaceValue) =>
+    val resolved = values.map { case (colName, replaceValue) =>
       // Check column name exists
-      df.resolve(colName)
+      val resolvedColumn = df.resolve(colName)
```

Comment:
Since the input string is the plain column name, we should call neither `df.resolve` nor `df.col`. We should follow `Dataset.drop/withColumn` to get the top-level attribute directly:
```
val columnEquals = df.sparkSession.sessionState.analyzer.resolver
val output = df.queryExecution.analyzed.output
val attrToValue = AttributeMap(values.map { case (colName, replaceValue) =>
  output.find(a => columnEquals(a.name, colName)).getOrElse(throw ...) -> replaceValue
})
val projections = output.map {  attr =>
  attrToValue.get(attr).map {
     case v: jl.Float => fillCol[Float](attr, v) // add an overload of fillCol that takes Attribute
     ...
  }.getOrElse(Column(attr))
}
...
```

Reply:
Above mentioned approach will not work if the null-fill map has qualified column name as below.
Eg: 
`def payment = spark.read.format("csv").option("header", "true")
      .schema(paymentSchema)
      .load("Payment.csv").as("payment")

    payment.na.fill(Map("payment.`Customer.Id`" -> -1))
      .show() `
Also, column name quoted with back-tick(`) will also not work.

Reply:
This is a good point, but we should forbid nested fields as we only support top-level columns here. How about
```
val attrToValue = AttributeMap(values.map { case (colName, replaceValue) =>
  val attr = df.resolve(colName) match {
    case a: Attribute => a
    case _ => throw new IllegalArgumentException("Nested field is not supported")
  }
  attr -> replaceValue
})
val projections = output.map {  attr =>
  attrToValue.get(attr).map {
     case v: jl.Float => fillCol[Float](attr, v) // add an overload of fillCol that takes Attribute
     ...
  }.getOrElse(Column(attr))
}
```

Reply:
Incorporated the suggested changes. Please review.